store-implementation/proxy-store/src/main/java/uk/gov/gchq/gaffer/proxystore/response/deserialiser/impl/DefaultResponseDeserialiser.java

```diff
@@ -36,8 +42,21 @@ public O deserialise(final String jsonString) throws SerialisationException {
             // The input is likely a plain java.lang.String object, so return as-is
             return (O) jsonString;
         } else {
-            // The input is likely a valid JSON value type, so process using the deserialiser
-            return JSONSerialiser.deserialise(encodeString(jsonString), typeReference);
+            // For some operations like 'Limit' we may not know the output type up front so it
+            // will be left as a generic '?', this means some objects will not be deserialised correctly.
+            try {
+                // Check if we can deserialise to common outputs like list of elements.
+                // This is a workaround until proxy store has been refactored
+                if (typeReference.getType().getTypeName().equals(Iterable.class.getName() + "<?>")) {
+                    return (O) JSONSerialiser.deserialise(encodeString(jsonString), new TypeReferenceImpl.IterableElement());
+                } else {
+                    return JSONSerialiser.deserialise(encodeString(jsonString), typeReference);
+                }
+            } catch (final SerialisationException e) {
+                // The input is likely a valid JSON value type, so process using the deserialiser
+                LOGGER.error("Unable to deserialse Iterable<?> as Iterable<Elements> using default deserialisation", e);
+                return JSONSerialiser.deserialise(encodeString(jsonString), typeReference);
```

Comment:
"try" at line 53 "return JSONSerialiser.deserialise(encodeString(jsonString), typeReference);"
"catch" line 58 "return JSONSerialiser.deserialise(encodeString(jsonString), typeReference);"

the scoping of this try catch does not seem correct.

Reply:
Was discussed in [a previous comment here](https://github.com/gchq/Gaffer/pull/3214#discussion_r1603322513). Ideally could be better but is only temporary until a better proxystore is done.